# HG changeset patch
# User Daniel Burrows <dburrows@debian.org>
# Date 1215965629 25200
# Node ID dcdb27036ceeac387dabfc3fd5fd935207a2c949
# Parent  657b68f8a501c16f3d5a4f4c675ec7bdcecdc677
Correctly handle install-and-mark-auto commands that are targeted at a particular version (e.g., "install foo/archive+M").
The version on the command-line was only being selected for an
"install" action; for "installauto" the code would set the candidate
version to be the current verison, with the result that nothing at all
would happen.

diff -r 657b68f8a501 -r dcdb27036cee src/cmdline/cmdline_action.cc
--- a/src/cmdline/cmdline_action.cc	Sat Jul 12 17:39:01 2008 -0700
+++ b/src/cmdline/cmdline_action.cc	Sun Jul 13 09:13:49 2008 -0700
@@ -292,7 +292,7 @@
     }
 
   pkgCache::VerIterator ver=pkg.CurrentVer();
-  if(action==cmdline_install)
+  if(action==cmdline_install || action == cmdline_installauto)
     ver=cmdline_find_ver(pkg, source, sourcestr);
 
   switch(action)
